components/service/glean/src/test/java/mozilla/components/service/glean/histogram/PrecomputedHistogramTest.kt

```diff
@@ -0,0 +1,137 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.histogram
+
+import androidx.test.core.app.ApplicationProvider
+import mozilla.components.service.glean.private.HistogramType
+import mozilla.components.service.glean.testing.GleanTestRule
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class PrecomputedHistogramTest {
+    @get:Rule
+    val gleanRule = GleanTestRule(ApplicationProvider.getApplicationContext())
+
+    @Test
+    fun `samples go in the correct bucket for exponential bucketing`() {
+        val td = PrecomputedHistogram(
+            rangeMin = 0L,
+            rangeMax = 60000L,
+            bucketCount = 100,
+            histogramType = HistogramType.Exponential
+        )
+
+        for (i in (0..60)) {
+            val x = i * 1000L
+            val bucket = td.findBucket(x)
+            assert(bucket <= x)
+        }
+
+        val td2 = PrecomputedHistogram(
+            rangeMin = 10000L,
+            rangeMax = 50000L,
+            bucketCount = 50,
+            histogramType = HistogramType.Exponential
+        )
+
+        for (i in (0..60)) {
+            val x = i * 1000L
+            val bucket = td2.findBucket(x)
+            assert(bucket <= x)
+        }
+    }
+
+    @Test
+    fun `validate the generated linear buckets`() {
+        val td = PrecomputedHistogram(
+            rangeMin = 0L,
+            rangeMax = 99L,
+            bucketCount = 100,
+            histogramType = HistogramType.Linear
+        )
+        assertEquals(100, td.buckets.size)
+
+        for (i in (0L until 100L)) {
+            val bucket = td.findBucket(i)
+            assert(bucket <= i)
+        }
+
+        val td2 = PrecomputedHistogram(
+            rangeMin = 50L,
+            rangeMax = 50000L,
+            bucketCount = 50,
+            histogramType = HistogramType.Linear
+        )
+        assertEquals(50, td2.buckets.size)
+
+        for (i in (0..60)) {
+            val x: Long = i * 1000L
+            val bucket = td2.findBucket(x)
+            assert(bucket <= x)
+        }
+    }
+
+    @Test
+    fun `toJsonObject and toJsonPayloadObject correctly converts a PrecomputedHistogram object`() {
+        // Define a PrecomputedHistogram object
+        val tdd = PrecomputedHistogram(
+            rangeMin = 0L,
+            rangeMax = 60000L,
+            bucketCount = 100,
+            histogramType = HistogramType.Exponential
+        )
+
+        // Accumulate some samples to populate sum and values properties
+        tdd.accumulate(1L)
+        tdd.accumulate(2L)
+        tdd.accumulate(3L)
+
+        // Convert to JSON object using toJsonObject()
+        val jsonTdd = tdd.toJsonObject()
+
+        // Verify properties
+        assertEquals("JSON bucket count must match Custom Distribution bucket count",
```

Comment:
Is the `Custom Distribution` the correct term here?

Reply:
Yeah -- should probably be "precomputed histogram"